test/unit/StakingPool/depositTo.js

```diff
@@ -0,0 +1,425 @@
+    };
+    await stakingPool.connect(this.coverSigner).allocateStake(coverRequest);
+
+    await increaseTime(
+      25 * // days
+        24 * // hours
+        60 * // minutes
+        60, // seconds
+    );
+
+    const firstDepositData = await stakingPool.deposits(depositNftId, firstActiveTrancheId);
+    const newStakeShares = await estimateStakeShares({ amount, stakingPool });
+
+    // deposit to the same tokenId
+    await stakingPool.connect(user).depositTo([
+      {
+        amount,
+        trancheId: firstActiveTrancheId,
+        tokenId: depositNftId,
+        destination,
+      },
+    ]);
+
+    const updatedDepositData = await stakingPool.deposits(depositNftId, firstActiveTrancheId);
+    const newRewardShares = await getNewRewardShares({
+      stakingPool,
+      initialStakeShares: firstDepositData.stakeShares,
+      stakeSharesIncrease: newStakeShares,
+      initialTrancheId: firstActiveTrancheId,
+      newTrancheId: firstActiveTrancheId,
+    });
+
+    // TODO: How can I make pendingRewards and lastAccNxmPerRewardShare to not be 0?
+    expect(updatedDepositData.pendingRewards).to.equal(0);
+    expect(updatedDepositData.lastAccNxmPerRewardShare).to.equal(0);
+    expect(updatedDepositData.stakeShares).to.equal(firstDepositData.stakeShares.add(newStakeShares));
+    expect(updatedDepositData.rewardsShares).to.equal(firstDepositData.rewardsShares.add(newRewardShares));
+  });
+
+  it('should not revert with division by zero', async function () {
+    const { stakingPool } = this;
+    const {
+      members: [user],
+    } = this.accounts;
+
+    const { amount, tokenId, destination, depositNftId } = depositToFixture;
+
+    const { firstActiveTrancheId } = await getTranches();
+
+    await stakingPool.connect(user).depositTo([
+      {
+        amount,
+        trancheId: firstActiveTrancheId,
+        tokenId,
+        destination,
+      },
+    ]);
+
+    // Generate rewards
+    const coverRequest = {
+      coverId: 0,
+      productId: 0,
+      amount: parseEther('1'),
+      period: 3600 * 24 * 30, // 30 days
+      gracePeriod: 3600 * 24 * 30,
+      globalCapacityRatio: 20000,
+      capacityReductionRatio: 0,
+      rewardRatio: 5000,
+    };
+    await stakingPool.connect(this.coverSigner).allocateStake(coverRequest);
+
+    // Multiple buckets/tranches expired?
+    await increaseTime(
+      150 * // days
+        24 * // hours
+        60 * // minutes
+        60, // seconds
+    );
+
+    // Reverts with Error: VM Exception while processing transaction:
+    // reverted with panic code 0x12 (Division or modulo division by zero)
+    // at StakingPool.updateTranches (contracts/modules/staking/StakingPool.sol:283)
+    // at StakingPool.depositTo (contracts/modules/staking/StakingPool.sol:353)
+    await stakingPool.connect(user).depositTo([
+      {
+        amount,
+        trancheId: firstActiveTrancheId,
+        tokenId: depositNftId,
+        destination,
+      },
+    ]);
```

Comment:
@shark0der @roxdanila This call is reverting with division by zero. Found it randomly when working on tests. Is this a valid scenario/bug?

Reply:
Seems to be a valid bug that we should address. Good catch! Please add a `.skip` for now and we'll open an issue for it.

Reply:
Went ahead and implemented a fix as it seemed simple enough to do it straight away. Now it reverts with `StakingPool: Requested tranche has expired`. Please take a look!

Reply:
Awesome! Thanks!